examples/plot_hawkes_basis_kernels.py

```diff
@@ -38,36 +38,63 @@ def g2(t):
     return np.cos(np.pi * (t / 10 + 1)) + 1.1
 
 
-t_values = np.linspace(0, 20, 1000)
```

Comment:
I thought it would give some structure. Moreover, one could open a python console, import the script, and run bits of the example by calling the functions with different parameters. 
However, I reverted it back to the imperative style.

Reply:
I prefer the more imperative style (actually I like mimicking the great work done by sklearn, cf https://scikit-learn.org/stable/auto_examples/index.html). That might be personal but I think keeping it for these few files would made them more coherent with the other examples in the project.

Reply:
Understood! It makes sense